packages/serverpod/lib/src/database/concepts/exceptions.dart

```diff
@@ -0,0 +1,44 @@
+/// Exception thrown when an error occurs in the database.
+abstract interface class DatabaseException implements Exception {
+  /// Returns the message of the exception.
+  String get message;
+
+  @override
+  String toString() => 'DatabaseException: $message';
+}
+
+/// Exception thrown when an exception occurs during a database query.
+abstract interface class DatabaseQueryException implements DatabaseException {
+  /// Returns the error code of the exception.
+  String? get code;
+
+  /// Additional details if provided by the database.
+  String? get detail;
+
+  /// A hint on how to remedy an error, if provided by the database.
+  String? get hint;
+
+  /// Returns the name of the table where the error occurred.
+  String? get tableName;
+
+  /// Returns the name of the column where the error occurred.
+  String? get columnName;
```

Comment:
Question: Should we make a distinction on that this class refers to relational databases and not document based one? If not, how do we see that potentially fit in in the future?

Reply:
Hmm, good question.

We could take a pragmatic approach here and treat the introduction of a document based database as a breaking change in the future? These values are nullable so if they are not used it should be fine.

This also kind of makes sense since our Expression queries are pretty much tailored toward a relational databases.

Reply:
I agree I think that makes sense!